consensus/src/chained_bft/chained_bft_smr_test.rs

```diff
@@ -76,22 +73,29 @@ impl SMRNode {
         let (commit_cb_sender, commit_cb_receiver) = mpsc::unbounded::<LedgerInfoWithSignatures>();
         let shared_mempool = MockSharedMempool::new(None);
         let consensus_to_mempool_sender = shared_mempool.consensus_sender.clone();
+        let state_computer = Arc::new(MockStateComputer::new(
+            state_sync_client,
+            commit_cb_sender,
+            Arc::clone(&storage),
+        ));
+        let txn_manager = Box::new(MockTransactionManager::new(Some(
+            consensus_to_mempool_sender,
+        )));
+        let (_tx, reconfig_events) =
+            libra_channel::new(QueueStyle::LIFO, NonZeroUsize::new(1).unwrap(), None);
 
-        let mut smr = ChainedBftSMR::new(
+        let input = ChainedBftSMRInput {
             network_sender,
             network_events,
-            &mut config.clone(),
-            Arc::new(MockStateComputer::new(
-                state_sync_client,
-                commit_cb_sender,
-                Arc::clone(&storage),
-                executor_with_reconfig,
-            )),
-            storage.clone(),
-            Box::new(MockTransactionManager::new(Some(
-                consensus_to_mempool_sender,
-            ))),
-        );
+            safety_rules_manager: SafetyRulesManager::new(&mut config.clone()),
```

Comment:
I think this will break cluster tests, this is how safety rules gets its keys which will not be properly cloned. Why do we want to clone this?

Reply:
this is just unit tests

Reply:
Hahaha, woops, I should have looked at the file name